### PR TITLE
Add a commit generation script

### PR DESCRIPTION
**Pull request by CameronDiver**
Change-type: minor
Signed-off-by: Cameron Diver <cameron@balena.io>

### CODE REVIEW

File: generate-commit.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+set -ex
+echo "Pulling meta-balena"
+
+if [ -z $TAG ]; then
+	echo "A \$TAG variable is required with the new supervisor version"
+	exit 1
+fi
+
+if [ ! -d /tmp/meta-balena ]; then
```
Comment (imrehg) [inline generate-commit.sh:13]: I would make `/tmp/meta-balena` a varible, and use that through the script. That might not be the place where people want to clone stuff, and would be good to be able to configure things

File: generate-commit.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+git pull
+git branch "supervisor-$TAG"
+git checkout "supervisor-$TAG"
+
+echo "Performing supervisor update"
+sed -i "s/SUPERVISOR_TAG.*/SUPERVISOR_TAG ?= \"${TAG}\"/g" meta-resin-common/recipes-containers/resin-supervisor/resin-supervisor.inc
+
+echo "Commiting and pushing"
+git add meta-resin-common/recipes-containers/resin-supervisor/resin-supervisor.inc
+git commit -s -F- <<EOF
```
Comment (imrehg) [inline generate-commit.sh:33]: would you want to add `-S` (signed commits), or you are not signing commits in general? :P

File: generate-commit.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+balena-supervisor: Update to ${TAG}
+
+Changelog-entry: Update balena-supervisor to ${TAG}
+Change-type: minor
```
Comment (imrehg) [inline generate-commit.sh]: Does `meta-balena` need the `change-type`? Is it in use?

File: generate-commit.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+if [ -z $TAG ]; then
+	echo "A \$TAG variable is required with the new supervisor version"
+	exit 1
+fi
+
+if [ ! -d /tmp/meta-balena ]; then
+	git clone git@github.com:balena-os/meta-balena /tmp/meta-balena
+else
+	cd /tmp/meta-balena
+	git reset --hard
```
Comment (imrehg) [inline generate-commit.sh:17]: also `git checkout master` before and `git pull` after, to work on the latest version?

**Review by imrehg — APPROVED** [review]
A couple of comments-slash-questions, but it seems usable as is.

File: generate-commit.sh (hunk 2 above)
Comment (CameronDiver) [reply generate-commit.sh:33]: Good point, it's in my gitconfig so I didn't notice it not being used.

File: generate-commit.sh (hunk 3 above)
Comment (CameronDiver) [reply generate-commit.sh]: Yup.

File: generate-commit.sh (hunk 4 above)
Comment (CameronDiver) [reply generate-commit.sh:17]: We do that after `development`, as that's what we merge the changes onto.

**Pull request merged**